pkg/alibabacloud/eni/node.go

```diff
@@ -93,7 +93,10 @@ func (n *Node) PopulateStatusFields(resource *v2.CiliumNode) {
 // attaches it to the instance as specified by the CiliumNode. neededAddresses
 // of secondary IPs are assigned to the interface up to the maximum number of
 // addresses as allowed by the instance.
-func (n *Node) CreateInterface(ctx context.Context, allocation *ipam.AllocationAction, scopedLog *logrus.Entry) (int, string, error) {
+func (n *Node) CreateInterface(ctx context.Context, allocation *ipam.AllocationAction, scopedLog *logrus.Entry, family ipam.Family) (int, string, error) {
+	if family == ipam.IPv6 {
```

Comment:
Does this mean it's possible to start Cilium operator with ipv6 and not have it fail until here?  In that case we should probably add a check at startup so that we fail before getting to this point.

Reply:
Commit f353505 implements this recommendation by adding IPAM provider and IPv6 checks to the operator and daemon commands.